### ISSUE TITLE
[Component] [tooltip, input] Tooltips absorb spaces when wrapping inputs

### ISSUE THREAD

**Issue by nietonfir** [element-plus/element-plus#8586]
Bug Type: **`Component`**

## Environment

- Vue Version: `3.2.36`
- Element Plus Version: `2.2.8`
- Browser / OS: `Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:101.0) Gecko/20100101 Firefox/101.0`
- Build Tool: `Vite`

### Related Component

- `el-tooltip`
- `el-input`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJBcHAudnVlIjoiPHNjcmlwdCBzZXR1cCBsYW5nPVwidHNcIj5cbmltcG9ydCB7IHJlZiwgdmVyc2lvbiBhcyB2dWVWZXJzaW9uIH0gZnJvbSAndnVlJ1xuaW1wb3J0IHsgdmVyc2lvbiBhcyBFcFZlcnNpb24gfSBmcm9tICdlbGVtZW50LXBsdXMnXG5pbXBvcnQgeyBFbGVtZW50UGx1cyB9IGZyb20gJ0BlbGVtZW50LXBsdXMvaWNvbnMtdnVlJ1xuXG5jb25zdCBtc2cgPSByZWYoJ0hlbGxvIFdvcmxkIScpXG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuICA8ZGl2PlxuICBcdDxoMT5lbC1pbnB1dCB3aXRoIHRvb2x0aXA8L2gxPlxuICAgIDxwPlxuICAgICAgUGxlYXNlIG5vdGUsIHRoYXQgcHJlc3Npbmcgc3BhY2UgaW4gdGhlIGlucHV0IGZpZWxkIHRyaWdnZXJzIHRoZSB0b29sdGlwIGFuZCBkb2VzIG5vdCBhZGQgYSBzcGFjZSB0byB0aGUgaW5wdXQuIChTdWJtaXR0aW5nIGEgZm9ybSB3aXRoIG9ubHkgb25lIGlucHV0IGZpZWxkIHdpbGwgcHJvYmFibHkgbm90IHdvcmsgYXMgd2VsbClcbiAgICA8L3A+XG4gIFx0PGVsLXRvb2x0aXAgOmNvbnRlbnQ9XCJtc2dcIj5cbiAgICBcdDxlbC1pbnB1dCB2LW1vZGVsPVwibXNnXCIgLz5cbiAgXHQ8L2VsLXRvb2x0aXA+XG4gICAgXG4gIDwvZGl2PlxuICBcbiAgPGRpdj5cbiAgXHQ8aDE+ZWwtaW5wdXQgKHdpdGhvdXQgdG9vbHRpcCk8L2gxPlxuICAgIDxlbC1pbnB1dCB2LW1vZGVsPVwibXNnXCIgLz5cbiAgPC9kaXY+XG4gIFxuXG4gIDxkaXY+XG4gIFx0PGgxPmlucHV0ICh3aXRob3V0IHRvb2x0aXApPC9oMT5cbiAgICA8ZWwtdG9vbHRpcCBjb250ZW50PVwiV3JhcHBpbmcgYSBzaW1wbGUgSFRNTDUgaW5wdXQgZWxlbWVudFwiPlxuXHQgICAgPGlucHV0IHR5cGU9XCJ0ZXh0XCIgLz5cbiAgICA8L2VsLXRvb2x0aXA+XG4gIDwvZGl2PlxuICBcbiAgPHA+XG4gICAgPGVsLWljb24gY29sb3I9XCJ2YXIoLS1lbC1jb2xvci1wcmltYXJ5KVwiPjxFbGVtZW50UGx1cyAvPjwvZWwtaWNvbj5cbiAgICBFbGVtZW50IFBsdXMge3sgRXBWZXJzaW9uIH19ICsgVnVlIHt7IHZ1ZVZlcnNpb24gfX1cbiAgPC9wPlxuPC90ZW1wbGF0ZT5cbiIsImltcG9ydF9tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHt9XG59IiwiX28iOnt9fQ==)

### Steps to reproduce

Wrap an input element with a tooltip:

```vue
<script setup lang="ts">
import { ref } from 'vue'

const msg = ref('Hello World!')
</script>

<template>
  <el-tooltip content="Wrapping a simple HTML5 input element">
    <input type="text" />
  </el-tooltip>
</template>
```

### What is Expected?

When entering a text with a space, I want the space to be visible in the input field, e.g. "Hello World!"

### What is actually happening?

The space is not being passed down to the input field: "HelloWorld!". Instead the tooltip appears.

### Additional comments

Setting `:trigger-keys="[]"` is a viable work-around as long as one is aware of this breaking change.
Relevant PR: https://github.com/element-plus/element-plus/pull/7946

**Comment by ayuan-gy** [comment]
![image](https://user-images.githubusercontent.com/82928935/177011222-49046a87-9c61-4223-be1d-d8853d986de9.png)
**What you see happens when you press the key contained in the triggerKeys array.**
![image](https://user-images.githubusercontent.com/82928935/177011395-a78382c6-547c-4e79-b8d3-95d67a316ca6.png)
**The default triggerKeys array contains Spaces. I wonder if this is a bug.**

**Comment by nietonfir** [comment]
I fail to understand why this was implemented in the first place without restricting the event handler (e.g. the wrapped element needs to have focus) or adding a dedicated trigger for it?
In my opinion this clearly counts as a bug, implementation as well as specification wise.

**Comment by baiwusanyu-c** [comment]
I don't think this is a bug, because when you focus on the `trigger` element, you can control the display of `tooltip` through the space key and enter key of the keyboard.You can use the following code to temporarily solve your problem.
````
<el-tooltip :content="msg" :triggerKeys="[]">
      <el-input v-model="msg" />
</el-tooltip>
````
I think `triggerkeys` can be provided to users as an API, allowing users to determine which keys can trigger the display and hiding of `tooltip`

**Comment by tolking** [comment]
> default behavior from what is typically expected from a tooltip would look like:
> <el-tooltip content="content" :trigger="['focus', 'hover']" :trigger-keys="[]">
> Problem is changing this default behavior would be a breaking change, that can't be done immediately.

There are a lot of components inside that use the Tooltip component, this part may be adjusted in the future.

The `trigger-keys` is good for pure keyboard users, maybe you can try using only keyboard action Tooltip or Select components.

**Comment by tolking** [comment]
> default behavior from what is typically expected from a tooltip would look like:
> <el-tooltip content="content" :trigger="['focus', 'hover']" :trigger-keys="[]">
> Problem is changing this default behavior would be a breaking change, that can't be done immediately.

There are a lot of components inside that use the Tooltip component, this part may be adjusted in the future.

The `trigger-keys` is good for pure keyboard users, maybe you can try using only keyboard action Tooltip or Select components.
